### PR TITLE
partner(google-vertex-ai): Add Custom User Agent to Vertex AI SDK

### PR DESCRIPTION
**Pull request by holtskinner**
- **Description:** Add custom user agent to Vertex AI SDK initialization. Allows API usage metrics collection.
  - Follow-up to #12168
- **Dependencies:** `google-cloud-aiplatform` to be updated in PR https://github.com/googleapis/python-aiplatform/pull/3261
  - Before Merging, Update `raise_vertex_import_error(minimum_expected_version: str = "1.38.0")` to the actual version once the SDK is updated.
  - https://pypi.org/project/google-cloud-aiplatform/

Tested locally successfully when installing from source PR.
 
DO NOT MERGE - Until https://github.com/googleapis/python-aiplatform/pull/3261 is Released and `minimum_expected_version` is updated to correct version.

### CODE REVIEW

**Comment by baskaryan** [comment]
cc @lkuligin

**Review by lkuligin — APPROVED** [review]

File: libs/community/langchain_community/utilities/vertexai.py [hunk 1]
```diff
@@ -80,6 +94,9 @@ def init_vertexai(
         credentials=credentials,
     )
 
+    _, user_agent = _get_user_agent(module)
+    initializer.global_config.append_user_agent(user_agent)
```
Comment (efriis) [inline libs/community/langchain_community/utilities/vertexai.py:98]: getting mypy errors that this `append_user_agent` function doesn't exist on `_Config`. Also causes the tests to fail, so seems like a real problem even after updating packages.

@holtskinner @lkuligin let me know how's best to fix!

Comment (efriis) [reply libs/community/langchain_community/utilities/vertexai.py:98]: Oh I'm silly - looks like it'll be in 42 when ready. Will convert this to a draft PR for now!

**Comment by efriis** [comment]
@lkuligin not sure if this one made it over to https://github.com/langchain-ai/langchain-google yet - should this be re-opened over there?

**Comment by holtskinner** [comment]
I think that https://github.com/langchain-ai/langchain-google/pull/31/ covers everything

**Pull request closed without merging**